internal/api/token_refresh.go

```diff
@@ -106,41 +106,64 @@ func (a *API) RefreshTokenGrant(ctx context.Context, w http.ResponseWriter, r *h
 			// refresh token row and session are locked at this
 			// point, cannot be concurrently refreshed
 
-			if token.Revoked {
-				a.clearCookieTokens(config, w)
-				// For a revoked refresh token to be reused, it
-				// has to fall within the reuse interval.
-
-				reuseUntil := token.UpdatedAt.Add(
-					time.Second * time.Duration(config.Security.RefreshTokenReuseInterval))
+			activeRefreshToken, terr := session.FindCurrentlyActiveRefreshToken(tx)
```

Comment:
seems like we only need to fetch the `activeRefreshToken` if the token is revoked, can we move this into the `if token.Revoked {}` block instead? 

would help to reduce an extra call to the database if the token is valid